### PR TITLE
FlatList functions scrollToEnd and scrollToIndex have been exposed. added setter for getItemLayout 

### PR DESCRIPTION
**Pull request by simonecorsato**
Hi, I just exposed a few functions of FlatList in order to be able to call them also on react-native-draggable-flatlist. Maybe someone else will find them useful.
Thanks

### CODE REVIEW

File: index.tsx [hunk 1]
```diff
@@ -510,6 +510,18 @@ class DraggableFlatList<T> extends React.Component<Props<T>, State> {
...
 
+  scrollToIndex = (index) => {
+    this.flatlistRef.current._component.scrollToIndex(index)
+  }
+
+  scrollToEnd = (params) => {
+		this.flatlistRef.current._component.scrollToEnd(params)
+  }	
+  
+  set getItemLayout(func) {
```
Comment (computerjazz) [inline index.tsx]: why doesn't this work as a prop?

File: index.tsx [hunk 2]
```diff
@@ -510,6 +510,18 @@ class DraggableFlatList<T> extends React.Component<Props<T>, State> {
     this.flatlistRef.current._component.scrollToOffset({ offset })
   })
 
+  scrollToIndex = (index) => {
+    this.flatlistRef.current._component.scrollToIndex(index)
+  }
+
+  scrollToEnd = (params) => {
+		this.flatlistRef.current._component.scrollToEnd(params)
```
Comment (computerjazz) [inline index.tsx]: fix indentation and add types

Comment (simonecorsato) [reply index.tsx]: I fixed indentation and added types. Is it ok?

File: index.tsx (hunk 1 above)
Comment (simonecorsato) [reply index.tsx]: I removed `set` keyword. Is there a better way to set `getItemLayout` property on FlatList or is it ok for you now?

File: index.tsx [hunk 3]
```diff
@@ -510,6 +520,18 @@ class DraggableFlatList<T> extends React.Component<Props<T>, State> {
...
 
+  scrollToIndex = (params: {animated: boolean, index: number, viewOffset: number, viewPosition: number}): void => {
+    this.flatlistRef.current._component.scrollToIndex(params)
+  }
+
+  scrollToEnd = (params: {animated: boolean}): void => {
+    this.flatlistRef.current._component.scrollToEnd(params)
+  }	
+    
+  getItemLayout = (func: GetItemLayoutFunction ): void => {
```
Comment (computerjazz) [inline index.tsx]: hm that's not quite what i meant. Props are spread onto the underlying flatlist. is there a reason that passing `getItemLayout` as a prop doesn't work, like in a normal flatlist? https://github.com/computerjazz/react-native-draggable-flatlist/blob/master/index.tsx#L802

Comment (simonecorsato) [reply index.tsx]: oh sorry, now I get what you mean. 
Anyway I checked the code that was using getItemLayout and I think it was legacy code not useful anymore. So I removed that part.

**Comment by billouboq** [comment]
why don't you just use "onRef" props ?

**Comment by simonecorsato** [comment]
The legacy code that was calling getItemLayout was using a mobx action and I think that was changing getItemLayout after that the props were already been passed to the list. I don't even know if the one who wrote the code reached his purpose, anyway it's no more required now, so I can remove that part.

**Comment by computerjazz** [comment]
@billouboq makes a great point -- both of these methods can be accessed via the FlatList ref exposed via the `onRef` prop. Closing this one but thanks for the PR!

**Pull request closed without merging**